frontend/packages/console-plugin-sdk/src/codegen/index.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-env node */
+
+import * as path from 'path';
+import * as readPkg from 'read-pkg';
+
+type Package = readPkg.NormalizedPackageJson;
+
+interface PluginPackage extends Package {
+  consolePlugin: {
+    entry: string;
+  }
+}
+
+function isValidPluginPackage(pkg: Package): pkg is PluginPackage {
+  return (pkg as PluginPackage).consolePlugin && typeof (pkg as PluginPackage).consolePlugin.entry === 'string';
+}
+
+function readPackages(packageFiles: string[]) {
+  const pkgList: Package[] = packageFiles.map(file => readPkg.sync({ cwd: path.dirname(file), normalize: true }));
+
+  return {
+    appPackage: pkgList.find(pkg => pkg.name === '@console/app'),
+    pluginPackages: pkgList.filter(isValidPluginPackage),
+  };
+}
+
+/**
+ * Generate the "active plugins" module source.
+ *
+ * @param packageFiles Paths to `package.json` files (all the monorepo packages).
+ */
+export function getActivePluginsModule(packageFiles: string[]): string {
+  const { appPackage, pluginPackages } = readPackages(packageFiles);
+  let output = `
+    var activePlugins = [];
```

Comment:
`var` => `const`
Even though I think the generated code is only used in a virtual file?

Reply:
Previously, the `@console/active-plugins` module source was generated as part of the webpack `afterCompile` hook, at which point the code was expected to be transformed, hence the use of `var` instead of `const`.

Currently, the `@console/active-plugins` module source is generated as part of webpack config object creation, i.e. ahead of webpack compiler execution, which means that we can use the modern ES (or TS) syntax.

Thanks for pointing this out. I'll change it to `const activePlugins`.

> Even though I think the generated code is only used in a virtual file?

Correct. The `getActivePluginsModule` function returns the source of `@console/active-plugins` module, which is a purely virtual module.